src/plugins/mer/meremulatordevicewidget.cpp

```diff
@@ -137,7 +139,7 @@ void MerEmulatorDeviceWidget::userNameEditingFinished()
         SshConnectionParameters sshParams = device->sshParameters();
         const QString& user = m_ui->userLineEdit->text();
         //TODO fix me:
-        const QString privKey = device->sharedConfigPath() +
+        const QString privKey = MerSdkManager::sdks().first()->sharedConfigPath() +
```

Comment:
`sdks()` returns non-const `QList` so here is a copy+detach.
`MerSdkManager::sdks().constFirst()` might be a bit faster but of course it is up to you to decide in this trade of readability vs a few CPU cycles optimization.

Reply:
Hm, it never occurred to me that `first()` is different from `at(0)`... that's really ugly. Even uglier it is with e.g. `QMap` because there is no `QMap::constFirst`, just `constBegin`. And what is the worst is that you cannot easily tell if it would actually detach without knowing or guessing the implementation of the function on the left side. Like in this particular case - it would actually not detach I think, because `sdks()` returns a temporary. So I will leave as is. Good to know - thanks.